### PR TITLE
Create main.yml

### PR DESCRIPTION
**Pull request by aguschirico**

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+        uses: actions/setup-node@v2
+        with:
+          node-version: '16'
+              
+      - name: Install dependencies
+        run: npm ci
+
+      - name: Deploy contracts
+        run: >
+          MNEMONIC=${{secrets.MNEMONIC}}
```
Comment (pmprete) [inline .github/workflows/main.yml:47]: As we have a default mnemonic i would remove this from the test
```suggestion
```

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+        with:
+          node-version: '16'
+
+      - name: Install dependencies
+        run: npm ci
+
+      - name: Tests
+        run: npm run test
+
+  deploy:
```
Comment (pmprete) [inline .github/workflows/main.yml:29]: I would remove the whole deploy task and instead would ad a step like:
- name: Deploy
        run: npm run deploy hardhat
        
 As it's currently creating another docker image just for the deploy and installing the dependecies

**Review by pmprete — APPROVED** [review]
Except my comments looks good

File: .github/workflows/main.yml (hunk 1 above)
Comment (aguschirico) [reply .github/workflows/main.yml]: done

File: .github/workflows/main.yml (hunk 2 above)
Comment (aguschirico) [reply .github/workflows/main.yml]: done

**Pull request merged**